apps/experiments/models.py

```diff
@@ -639,8 +640,48 @@ def send_bot_message(self, instruction_prompt: str, fail_silently=True):
         bot_message = bot_prompt_for_user(self, prompt_instruction=instruction_prompt)
         try_send_message(self, message=bot_message, fail_silently=fail_silently)
 
+    def get_participant_scheduled_messages(self, as_dict=False):
+        """
+        Returns all scheduled messages for the associated participant for this session's experiment as well as
+        any child experiments in the case where the experiment is a parent
+
+        Parameters:
+        as_dict: If True, the data will be returned as an array of dictionaries, otherwise an an array of strings
+        """
+        from apps.events.models import ScheduledMessage
+
+        child_experiments = ExperimentRoute.objects.filter(team=self.team, parent=self.experiment).values("child")
+        messages = ScheduledMessage.objects.filter(
+            Q(experiment=self.experiment) | Q(experiment__in=models.Subquery(child_experiments)),
+            participant=self.participant,
+            team=self.team,
+        ).select_related("action")
+
+        scheduled_messages = []
+        for message in messages:
+            if as_dict:
+                scheduled_messages.append(
+                    {
+                        "name": message.name,
+                        "frequency": message.frequency,
+                        "time_period": message.time_period,
+                        "repetitions": message.repetitions,
+                        "next_trigger_date": pretty_date(message.next_trigger_date),
```

Comment:
nit: this should be isoformat
```suggestion
                        "next_trigger_date": message.next_trigger_date.isoformat(),
```

Reply:
I was gonna ask why, but then realized that in order to reasily work with it, it should be. Good catch